pages/api/messageTemplates/[[...id]].ts

```diff
@@ -0,0 +1,85 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { getMockTemplates } from './getAllTemplates'
+import { getMockTemplate } from './getTemplate'
+import { createMockTemplate, createMockTemplateWithId } from './createTemplate'
+import { updateMockTemplate } from './updateTemplate'
+import { deleteMockTemplate } from './deleteTemplate'
+
+export default async function handler (
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const { id } = req.query
+  const method = req.method
+
+  if (typeof id === 'string' || !isValidRouteIdArr(id, method)) {
+    res.status(400).json(JSON.stringify(
+      'Bad request: Bad number of id\'s specified in route.'
+    ))
+    return
+  }
+
+  let status: number
+  let message: any
+  switch (method) {
+    case undefined:
+      status = 400
+      message = 'HTTP method must be explicitly defined'
+      break
+    case 'GET':
+      if (!id) {
+        status = 200
+        message = getMockTemplates()
+      } else {
+        status = 200
+        message = getMockTemplate(parseInt(id[0]))
+      }
+      break
+    case 'POST':
+      if (!(req.body.title && req.body.message && req.body.createdBy)) {
```

Comment:
Let's move the logic for each case into its own function since we'll definitely be adding more logic as we continue to build on top of this. Plopping the applicants endpoint from the app portal here for reference! 
<img width="840" alt="Screen Shot 2022-08-06 at 6 29 09 PM" src="https://user-images.githubusercontent.com/59738880/183267871-18033f7a-8a21-4ed0-b36c-56798083b694.png">